### DISCUSSION TITLE
clear Linux cache before checksum verify on USB key

### DISCUSSION THREAD

**Discussion by ebbi2017** [doublecmd/doublecmd#1095] [Suggestions]
### Description

1. Successfully downloaded a big iso file of f.e. Ubuntu live and checksum verify is okay.
2. Copied file onto USB stick with either Windows or Linux, shutdown too early as it seems (but copy file dialog was gone before).
3. Checksum verify after file copy is fery fast (reads from cache).
4. Reboot Ventoy, started new iso, Ubuntu says checksum is okay (which isn't as tested after reboot).
5. Reboot into main OS and checked checksum from USB key, didn't match.
6. Re-download onto internal SSD, checked checksum, which is okay.
7. Compared files on USB key and internal SSD. Some parts at the end are zeros, not yet copied, but copy file dialog was gone before shutdown.

Is it possible to add a cache clean button next to the checksum verify, so we can optionally do this before a checksum test?
I could clean the cache in Linux and indicators in my taskbar and "free" command show the differences.

Or is it somehow possible to just ignore the OS caches and read the real file instead?

### Proposed technical implementation details (optional)

sync; echo 1 | sudo tee /proc/sys/vm/drop_caches

or

sudo -s
sync; echo 1 > /proc/sys/vm/drop_caches

solves the problem, that the verify command uses the OS caches instead of reading the real file.

**Comment by alexx2000** [comment]
You can copy file to USB stick with "Verify" [option enabled](https://doublecmd.github.io/doc/en/copymove.html) in this case.

> Or is it somehow possible to just ignore the OS caches and read the real file instead?

Yes, it is need to open file with O_SYNC | O_DIRECT flags.
